### PR TITLE
Add Yacht exercise to the Scala track

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Per my [discussion thread](https://forum.exercism.org/t/yacht-exercise-not-available-in-scala-track/4568) on the exercism forum.

### CODE REVIEW

**Comment by wangxiaoxuan273** [comment]
@ErikSchierboom Could you please take a look at this pull request? I did not build, run or locally tested anything, since I did not find any guide about how to do any of these, so this PR may need further change.

I mainly copied the stuff from the F# Yacht problem and translated everything into Scala 2.

File: exercises/practice/yacht/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "authors": [
+    "ErikSchierboom"
+  ],
+  "contributors": [
+    "leafdulce"
+  ],
```
Comment (ErikSchierboom) [inline exercises/practice/yacht/.meta/config.json:7]:
```suggestion
  "authors": [
    "wangxiaoxuan273"
  ],
```

File: exercises/practice/yacht/src/test/scala/YachtTest.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import org.scalatest.matchers.should.Matchers
+
+class YachtTests extends AnyFunSuite with Matchers {
+
+  test("Yacht") {
+    Yacht.score(Category.Yacht, List(Die.Five, Die.Five, Die.Five, Die.Five, Die.Five)) shouldEqual 50
+  }
+
+  test("Not Yacht") {
+    Yacht.score(Category.Yacht, List(Die.One, Die.Three, Die.Three, Die.Two, Die.Five)) shouldEqual 0
```
Comment (ErikSchierboom) [inline exercises/practice/yacht/src/test/scala/YachtTest.scala:11]: All but the first test should be `pending`:

```suggestion
    pending
    Yacht.score(Category.Yacht, List(Die.One, Die.Three, Die.Three, Die.Two, Die.Five)) shouldEqual 0
```

Could you also update the other tests?

**Comment by ErikSchierboom** [comment]
configlet fails with:

> `/home/runner/work/scala/scala/exercises/practice` contains a directory named `yacht`, which is not a `slug` in the array of practice exercises. Please add the exercise to that array. If the exercise is not ready to be shown on the website, please set its `status` value to "wip":

You should add an entry for this exercise to the `exercises.practice` array in the [`config.json` file](https://github.com/exercism/scala/blob/main/config.json). Note that the uuid must be uniquely generated (you can do this via `bin/fetch-configlet && bin/configlet uuid`.

I'd also recommend update the tests, metadata and docs with the latest version: `bin/fetch-configlet && bin/configlet sync --update --yes --tests include --docs --metadata`.

All in all this is a great PR!

File: exercises/practice/yacht/src/test/scala/YachtTest.scala (hunk 2 above)
Comment (wangxiaoxuan273) [reply exercises/practice/yacht/src/test/scala/YachtTest.scala:12]: Added `pending` for the tests.

**Comment by wangxiaoxuan273** [comment]
> configlet fails with:
> 
> > `/home/runner/work/scala/scala/exercises/practice` contains a directory named `yacht`, which is not a `slug` in the array of practice exercises. Please add the exercise to that array. If the exercise is not ready to be shown on the website, please set its `status` value to "wip":
> 
> You should add an entry for this exercise to the `exercises.practice` array in the [`config.json` file](https://github.com/exercism/scala/blob/main/config.json). Note that the uuid must be uniquely generated (you can do this via `bin/fetch-configlet && bin/configlet uuid`.
> 
> I'd also recommend update the tests, metadata and docs with the latest version: `bin/fetch-configlet && bin/configlet sync --update --yes --tests include --docs --metadata`.
> 
> All in all this is a great PR!

Done these steps! When `bin/fetch-configlet && bin/configlet sync --update --yes --tests include --docs --metadata`, the program looked for `config.json` in the `bin`. I copied the file from the root directory, and did not commit it after the program had completed.

**Comment by wangxiaoxuan273** [comment]
@ErikSchierboom I have updated this pr according to your comments. Please take a look again, thanks.

**Comment by ErikSchierboom** [comment]
> Done these steps! When running bin/fetch-configlet && bin/configlet sync --update --yes --tests include --docs --metadata, the program looked for config.json in bin and could not find it. I copied the file from the root directory to bin to let the program run, and did not commit it after the program had completed.

Are you sure? configlet expects `config.json` in the root. Did you run it from the root? I also forgot that `--exercise yacht` should be added to the command. Could you redo the last commit by running the previous command but with `--exercise yacht` added?

**Comment by wangxiaoxuan273** [comment]
> > Done these steps! When running bin/fetch-configlet && bin/configlet sync --update --yes --tests include --docs --metadata, the program looked for config.json in bin and could not find it. I copied the file from the root directory to bin to let the program run, and did not commit it after the program had completed.
> 
> Are you sure? configlet expects `config.json` in the root. Did you run it from the root? I also forgot that `--exercise yacht` should be added to the command. Could you redo the last commit by running the previous command but with `--exercise yacht` added?

Sure! I did not run it in the root, I ran it inside the `bin`. It looks like the program looked for the config with a relative path.

**Comment by wangxiaoxuan273** [comment]
I have updated this pr according to your latest comments. Please take a look. @ErikSchierboom

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Comment by wangxiaoxuan273** [comment]
The CI tests failed. I'll work on fixing my errors.

**Comment by wangxiaoxuan273** [comment]
I fixed my error of wrong class name. Could you please start the CI test again? Thanks. @ErikSchierboom

**Comment by ErikSchierboom** [comment]
Thanks!

**Pull request merged**